src/Sulu/Bundle/AdminBundle/Resources/js/stores/MetadataStore/MetadataStore.js

```diff
@@ -1,24 +1,62 @@
 // @flow
-import {Requester} from '../../services';
+import symfonyRouting from 'fos-jsrouting/router';
+import {buildQueryString} from '../../utils/Request';
+
+const defaultOptions = {
+    credentials: 'same-origin',
+    headers: {
+        'Content-Type': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest',
+    },
+};
 
 class MetadataStore {
-    endpoint: string;
-    metadataPromises: {[string]: {[string]: Promise<Object>}} = {};
+    metadataPromises: {[string]: {[string]: ?Promise<Object>}} = {};
+
+    loadMetadata(type: string, key: string, metadataOptions: Object = {}): Promise<Object> {
+        const parameters = {
+            type: type,
+            key: key,
+            ...metadataOptions,
+        };
 
-    loadMetadata(type: string, key: string): Promise<Object> {
         if (!this.metadataPromises[type]) {
             this.metadataPromises[type] = {};
         }
+        const keyWithOptions = buildQueryString({
+            key: key,
+            ...metadataOptions,
+        });
+
+        if (!this.metadataPromises[type][keyWithOptions]) {
+            const url = symfonyRouting.generate('sulu_admin.metadata', parameters);
+            const response = fetch(url, defaultOptions).then((response) => {
+                if (!response.ok) {
+                    this.metadataPromises[type][keyWithOptions] = undefined;
+                    return Promise.reject(response);
+                }
+
+                if (response.status === 204) {
+                    this.metadataPromises[type][keyWithOptions] = undefined;
+                    return Promise.resolve({});
+                }
+
+                const cacheControl = response.headers.get('cache-control');
+                if (cacheControl && cacheControl.includes('no-store')) {
+                    this.metadataPromises[type][keyWithOptions] = undefined;
+                }
+
+                return response.json()
+                    .then((data) => {
+                        return data;
+                    });
+            });
 
-        if (!this.metadataPromises[type][key]) {
-            this.metadataPromises[type][key] = Requester.get(
-                this.endpoint
-                    .replace(':type', type)
-                    .replace(':key', key)
-            );
+            this.metadataPromises[type][keyWithOptions] = response;
+            return response;
```

Comment:
This line could just be removed, then the next `return` handles this the exact same way.

Reply:
That didn't work out, because it is not typesafe.